max/python/max/pipelines/architectures/gemma3multimodal/model_config.py

```diff
@@ -0,0 +1,300 @@
+    """Number of channels in the input images."""
+
+    patch_size: int
+    """The size (resolution) of each patch.  default 14"""
+
+    attention_bias: bool = True
+
+    attention_dropout: float = 0.0
+    """The dropout ratio for the attention probabilities"""
+
+    vision_use_head: bool = False
+    """Flag whether to use attention heads for vision"""
+
+    _HIDDEN_ACTIVATION_MAP = {
+        "gelu_pytorch_tanh": "tanh",
+        "swish": "silu",
+    }
+
+    @staticmethod
+    def generate(vision_config: AutoConfig) -> Gemma3VisionConfig:
+        hidden_act = vision_config.hidden_act
+        if hidden_act in Gemma3VisionConfig._HIDDEN_ACTIVATION_MAP:
+            hidden_act = Gemma3VisionConfig._HIDDEN_ACTIVATION_MAP[hidden_act]
+
+        return Gemma3VisionConfig(
+            hidden_size=vision_config.hidden_size,
+            image_size=vision_config.image_size,
+            intermediate_size=vision_config.intermediate_size,
+            num_attention_heads=vision_config.num_attention_heads,
+            num_hidden_layers=vision_config.num_hidden_layers,
+            patch_size=vision_config.patch_size,
+            num_channels=vision_config.num_channels,
+            hidden_act=hidden_act,
+            layer_norm_eps=vision_config.layer_norm_eps,
+        )
+
+
+@dataclass
+class Gemma3MultiModalConfigBase(MAXModelConfigBase):
+    """Base configuration for Gemma 3 models.
+
+    Contains parameters specific to the Gemma 3 architecture, typically
+    extracted from a HuggingFace configuration object's text config.
+    """
+
+    boi_token_index: int
+    """The begin-of-image token index to wrap the image prompt"""
+
+    eoi_token_index: int
+    """The end-of-image token index to wrap the image prompt"""
+
+    devices: list[DeviceRef]
+    """Devices to run the model with."""
+
+    dtype: DType
+    """DType of the model weights and input."""
+
+    kv_params: KVCacheParams
+    """KV cache parameters."""
+
+    image_token_index: int
+    """The image token index to encode the image prompt"""
+
+    initializer_range: float
+    """Standard deviation for weight initialization."""
+
+    interleaved_rope_weights: bool
+    """True if the rope weights are in interleaved complex format."""
+
+    mm_tokens_per_image: int
+    """The number of tokens per image embedding"""
+
+    return_logits: ReturnLogits
+    """Whether to return the last token, all logits, or a variable number of logits."""
+
+    tie_word_embeddings: bool
+    """Whether to tie weight embeddings. When true, the output linear layer
+    uses the same
+    weight as the embedding layer."""
+
+    text_config: Gemma3Config
+    """The config object of the text backbone"""
+
+    vision_config: Gemma3VisionConfig
+    """Custom vision config or dict"""
+
+    attention_bias: bool = False
+    """Whether to use a bias in the query, key, value and output projection layers during self-attention."""
+
+    attn_logit_softcapping: float = None
```

Comment:
It looks like this isn't used (and I don't think we support softcapping), perhaps this should be removed